### PR TITLE
chore: update deps

### PR DESCRIPTION
**Pull request by lucacasonato**
Updates a lot of dependencies. Skipped of `ext/crypto` as too much has changed
there to be able to bunch that into this PR.

Note: there is a duplicate `dprint-core` in here. @dsherret can you dedupe?

### CODE REVIEW

File: cli/Cargo.toml [hunk 1]
```diff
@@ -38,11 +38,11 @@ deno_websocket = { version = "0.54.0", path = "../ext/websocket" }
 deno_webstorage = { version = "0.44.0", path = "../ext/webstorage" }
 regex = "=1.5.5"
 serde = { version = "=1.0.136", features = ["derive"] }
-zstd = '=0.9.2'
+zstd = '0.11.1'
```
Comment (kitsonk) [inline cli/Cargo.toml:41]: Why not pinned?

File: cli/Cargo.toml [hunk 2]
```diff
@@ -57,49 +57,49 @@ atty = "=0.2.14"
...
 data-url = "=0.1.1"
-dissimilar = "=1.0.2"
+dissimilar = "=1.0.3"
 dprint-plugin-json = "=0.15.1"
 dprint-plugin-markdown = "=0.13.1"
-dprint-plugin-typescript = "=0.67.1"
-encoding_rs = "=0.8.29"
-env_logger = "=0.8.4"
+dprint-plugin-typescript = "0.67.1"
+encoding_rs = "0.8.31"
```
Comment (dsherret) [inline cli/Cargo.toml:68]: Add back equals signs?

**Review by dsherret — APPROVED** [review]
LGTM

File: Cargo.lock [hunk 3]
```diff
@@ -215,22 +215,25 @@ source = "registry+https://github.com/rust-lang/crates.io-index"
...
 
 [[package]]
 name = "autocfg"
-version = "0.1.7"
+version = "0.1.8"
 source = "registry+https://github.com/rust-lang/crates.io-index"
-checksum = "1d49d90015b3c36167a20fe2810c5cd875ad504b39cff3d4eae7977e6b7c1cb2"
+checksum = "0dde43e75fd43e8a1bf86103336bc699aa8d17ad1be60c76c0bdfd4828e19b78"
+dependencies = [
+ "autocfg 1.1.0",
```
Comment (bartlomieju) [inline Cargo.lock:229]: This is very strange... how come `autocfg 0.1.8` depends on `autocfg 1.1.0`?

**Review by bartlomieju — APPROVED** [review]
LGTM

File: Cargo.lock (hunk 3 above)
Comment (littledivy) [reply Cargo.lock:229]: https://docs.rs/autocfg/0.1.8/autocfg/

> As of autocfg 0.1.8, this is simply re-exporting from autocfg 1.1.0 or later. Please upgrade and refer to the newer version for current documentation!

Comment (bartlomieju) [reply Cargo.lock:229]: Joyful

File: cli/Cargo.toml [hunk 4]
```diff
@@ -111,8 +111,8 @@ flaky_test = "=0.1.0"
 os_pipe = "=1.0.1"
 pretty_assertions = "=1.2.1"
 test_util = { path = "../test_util" }
-trust-dns-client = "=0.21.2"
-trust-dns-server = "=0.21.2"
+trust-dns-client = "0.21.2"
+trust-dns-server = "0.21.2"
```
Comment (bartlomieju) [inline cli/Cargo.toml:115]: I think I made a mistake landing a PR, can you pin these?

**Pull request merged**